test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -0,0 +1,148 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.HeadlessException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+
+import static javax.swing.SwingUtilities.invokeAndWait;
+import static javax.swing.SwingUtilities.isEventDispatchThread;
+
+public class PassFailJFrame extends JFrame {
+    private final CountDownLatch latch = new CountDownLatch(1);
+    private boolean failed = false;
+    private String testFailedReason;
+    private JTextArea instructionsText;
+    private final int maxRowLength;
+    private final int maxStringLength;
+    private final int timeoutMinutes;
+
+    /**
+     * Constructs a JFrame with a given title & serves as test instructional
+     * frame where the user follows the specified test instruction in order
+     * to test the test case & mark the test pass or fail. If the expected
+     * result is seen then the user click on the 'Pass' button else click
+     * on the 'Fail' button and the reason for the failure should be
+     * specified in the JDailog JTextArea.
+     *
+     * @param title           title of the Frame.
+     * @param instructions    specified instruction that user should follow.
+     * @param maxRowLength    number of visible rows of the JTextArea where the
+     *                        instruction is show.
+     * @param maxStringLength number of columns of the instructional JTextArea
+     * @param timeoutMinutes  timeout of the test where time is specified in
+     *                        minutes.
+     * @throws HeadlessException
+     * @throws InterruptedException
+     * @throws InvocationTargetException
+     */
+    public PassFailJFrame(String title, String instructions,
+                          int maxRowLength, int maxStringLength,
+                          int timeoutMinutes) throws HeadlessException,
+            InterruptedException, InvocationTargetException {
+        super(title);
+        this.maxRowLength = maxRowLength;
+        this.maxStringLength = maxStringLength;
+        this.timeoutMinutes = timeoutMinutes;
+
+        invokeAndWait(() -> {
```

Comment:
This should also be preceded by a check as to whether it is EDT - I thought I suggested something along those lines already - make it so it works either way and the test itself doesn't have to care to make sure it is on or off EDT, and also even it if uses invokeAndWait when it didn't need to it is harmless.

Reply:
Added a check whether the method is called with in EDT else invoke the UI with in EDT.